test/cartesian/ReferenceLine/ReferenceLine.spec.tsx

```diff
@@ -268,80 +268,71 @@ describe('<ReferenceLine />', () => {
     });
   });
 
-  test('throws an error when rendered without any axes', () => {
-    expect(() => {
-      render(
-        <BarChart width={1100} height={250}>
-          <ReferenceLine />
-        </BarChart>,
-      );
-    }).toThrowError('Invariant failed: Could not find xAxis by id "0" [number]. There are no available ids.');
+  it('should not render anything when used without any axes', () => {
+    const { container } = render(
+      <BarChart width={1100} height={250}>
+        <ReferenceLine />
+      </BarChart>,
+    );
+    expect(container.querySelectorAll('.recharts-reference-line-line')).toHaveLength(0);
   });
 
-  test('throws an error when rendered without YAxis', () => {
-    expect(() => {
-      render(
-        <BarChart width={1100} height={250}>
-          <XAxis />
-          <ReferenceLine x={20} />
-        </BarChart>,
-      );
-    }).toThrowError('Invariant failed: Could not find yAxis by id "0" [number]. There are no available ids.');
+  it('should not render anything when rendered without YAxis', () => {
+    const { container } = render(
+      <BarChart width={1100} height={250}>
+        <XAxis />
+        <ReferenceLine x={20} />
+      </BarChart>,
+    );
+    expect(container.querySelectorAll('.recharts-reference-line-line')).toHaveLength(0);
   });
 
-  test('throws an error when rendered without XAxis', () => {
-    expect(() => {
-      render(
-        <BarChart width={1100} height={250}>
-          <YAxis />
-          <ReferenceLine y={20} />
-        </BarChart>,
-      );
-    }).toThrowError('Invariant failed: Could not find xAxis by id "0" [number]. There are no available ids.');
+  it('should not render anything when rendered without XAxis', () => {
+    const { container } = render(
+      <BarChart width={1100} height={250}>
+        <YAxis />
+        <ReferenceLine y={20} />
+      </BarChart>,
+    );
+    expect(container.querySelectorAll('.recharts-reference-line-line')).toHaveLength(0);
   });
 
-  test('throws when passed in invalid xAxisId', () => {
-    expect(() => {
-      render(
-        <BarChart width={1100} height={250}>
-          <XAxis />
-          <YAxis />
-          <ReferenceLine xAxisId="this ID definitely does not exist anywhere" />
-        </BarChart>,
-      );
-    }).toThrowError(
-      'Invariant failed: Could not find xAxis by id "this ID definitely does not exist anywhere" [string]. Available ids are: 0',
+  it('should not render anything when passed in invalid xAxisId', () => {
+    const { container } = render(
+      <BarChart width={1100} height={250}>
+        <XAxis />
+        <YAxis />
+        <ReferenceLine xAxisId="this ID definitely does not exist anywhere" />
```

Comment:
Shouldn't we still invariant on these? Now we're back to someone not knowing why their stuff doesn't render

Reply:
I was considering a more hollistic approach. Something like:
- all components should be permissive by default - if something is wrong, return undefined, never throw
- there should be a "strict mode" prop where each of these invariants will throw an error. Like, when chart width is zero.

What do you think?

Reply:
Personally if it's not working I think I'd like to know about it no matter what, but I do like the idea of modes. 

Sounds good then! We can make an issue to track that